svc-events/events/evtsubscription.go

```diff
@@ -898,12 +900,43 @@ func validateFields(request *evmodel.RequestBody) (int32, string, []interface{},
 		}
 	}
 	if request.EventFormatType == "" {
-		request.EventFormatType = evmodel.EventFormatType
-	} else if request.EventFormatType == "MetricReport" {
-		return http.StatusBadRequest, errResponse.PropertyMissing, []interface{}{"EventFormatType"}, fmt.Errorf("Unsupported EventFormatType")
-	} else if request.EventFormatType != evmodel.EventFormatType {
 		return http.StatusBadRequest, errResponse.PropertyMissing, []interface{}{"EventFormatType"}, fmt.Errorf("Invalid EventFormatType")
 	}
+	validFormat := false
+	for _, eventFormat := range validEventFormatTypes {
```

Comment:
could have used a map instead of an array, to check the EventFormatType present.